src/Common/src/Interop/Unix/libssl/SslConnectionInfo.cs

```diff
@@ -190,5 +202,32 @@ private static HashAlgorithmType MapHashAlgorithmType(Interop.libssl.DataHashAlg
                     return HashAlgorithmType.None;
             }
         }
+        
+        private static int GetHashAlgorithmBitSize(Interop.libssl.DataHashAlgorithm mac)
+        {
+            switch (mac)
+            {
+                case Interop.libssl.DataHashAlgorithm.SSL_MD5:
+                    return MD5_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA1:
+                    return SHA1_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST94:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_GOST89MAC:
+                    return GOST_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA256:
+                    return SHA256_HashKeySize;
+
+                case Interop.libssl.DataHashAlgorithm.SSL_SHA384:
+                    return SHA384_HashKeySize;
+
+                default:
+                    return 0;
```

Comment:
@CIPop wrote a little tool for printing the details of an SSL connection.

On the server side I ran `openssl s_server -cert eccombined.pem -cipher ECDHE-ECDSA-AES128-GCM-SHA256 -accept 8000`, forcing it into a GCM connection.  The Windows connection information says
```
        PROTOCOL=Tls12 , KEYEXCHG=( 256bit      44550), CYPHER=( 128bit     Aes128), HASH=(   0bit      32780)
```

I boosted it a bit:
```
 KeyEx: Class=5 Type=7 Sid=6
Cipher: Class=3 Type=3 Sid=14
  Hash: Class=4 Type=0 Sid=12
```

The hash appears to be reported as a 0-bit strong hash (class=4) any (type=0) SHA-256 (sid=12).   The relevant part for this function is that it agrees that the hash size should be 0 for AEAD.

What's probably better is to report 0 for AEAD (`# define SSL_AEAD 0x00000040L`) and -1 or throw, or something on otherwise unknown inputs.